spring-integration-file/src/main/java/org/springframework/integration/file/remote/session/ContextHolderRequestHandlerAdvice.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2015-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.file.remote.session;
+
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import org.springframework.messaging.Message;
+
+/**
+ * Provides a key for the context holder.This key could for example be stored in a message header.
+ *
+ * @author Adel Haidar
+ * @since 6.0
+ */
+public class ContextHolderRequestHandlerAdvice {
```

Comment:
You are missing an implementation of `AbstractRequestHandlerAdvice`.
Otherwise this class have no value.